scripts/commerce.js

```diff
@@ -559,11 +559,15 @@ function createHashFromObject(obj, length = 5) {
 
 /**
  * Creates a commerce endpoint URL with query parameters including a cache-busting hash.
+ * @param {string} [customerGroupId] - Optional customer group ID to include in the CB Hash
  * @returns {Promise<URL>} A promise that resolves to the endpoint URL with query parameters
  */
-export async function commerceEndpointWithQueryParams() {
+export async function commerceEndpointWithQueryParams(customerGroupId) {
   const urlWithQueryParams = new URL(getConfigValue('commerce-endpoint'));
   const headers = getHeaders('cs');
+  if (customerGroupId && headers['Magento-Customer-Group']) {
+    headers['Magento-Customer-Group'] = customerGroupId;
+  }
```

Comment:
I would change this signature to accept an object. It would be more flexible and allow for more customization, such as if a different header or key is needed. Also not sure why this function is async.

```suggestion
export function commerceEndpointWithQueryParams(customHeaders = {}) {
  const urlWithQueryParams = new URL(getConfigValue('commerce-endpoint'));
  const headers = {
    ...getHeaders('cs');
    ...customHeaders
  }
```

and in use:

```js
const endpoint = await commerceEndpointWithQueryParams({ 
  'Magento-Customer-Group': 'abc-123'
})
```

Reply:
Actually, headers here are just used to hash. It's not really necessary to add in customerGroupId.

Reply:
As discussed in slack - we can leave `async` change out of the changes here. I'll remove it from my suggestion.